common/src/main/java/org/corfudb/common/metrics/micrometer/registries/LoggingMeterRegistryWithHistogramSupport.java

```diff
@@ -83,130 +157,22 @@ protected void publish() {
-                                    if (!config.logInactive() && count == 0) return;
-                                    loggingSink.accept(print.id() + " throughput=" + print.unitlessRate(count) +
-                                            " mean=" + print.time(snapshot.mean(getBaseTimeUnit())) +
-                                            " max=" + print.time(snapshot.max(getBaseTimeUnit())));
-                                },
-                                summary -> {
-                                    HistogramSnapshot snapshot = summary.takeSnapshot();
-                                    long count = snapshot.count();
-                                    if (!config.logInactive() && count == 0) return;
-                                    loggingSink.accept(print.id() + " throughput=" + print.unitlessRate(count) +
-                                            " mean=" + print.value(snapshot.mean()) +
-                                            " max=" + print.value(snapshot.max()));
-                                },
-                                longTaskTimer -> {
-                                    int activeTasks = longTaskTimer.activeTasks();
-                                    if (!config.logInactive() && activeTasks == 0) return;
-                                    loggingSink.accept(print.id() +
-                                            " active=" + print.value(activeTasks) +
-                                            " duration=" + print.time(longTaskTimer.duration(getBaseTimeUnit())));
-                                },
-                                timeGauge -> {
-                                    double value = timeGauge.value(getBaseTimeUnit());
-                                    if (!config.logInactive() && value == 0) return;
-                                    loggingSink.accept(print.id() + " value=" + print.time(value));
-                                },
-                                counter -> {
-                                    double count = counter.count();
-                                    if (!config.logInactive() && count == 0) return;
-                                    loggingSink.accept(print.id() + " throughput=" + print.rate(count));
-                                },
-                                timer -> {
-                                    double count = timer.count();
-                                    if (!config.logInactive() && count == 0) return;
-                                    loggingSink.accept(print.id() + " throughput=" + print.rate(count) +
-                                            " mean=" + print.time(timer.mean(getBaseTimeUnit())));
-                                },
-                                meter -> loggingSink.accept(writeMeter(meter, print))
-                        );
-                    });
+                    .flatMap(meter -> meter.match(
+                            gauge -> writeGauge(gauge.getId(), gauge.value()),
```

Comment:
Does `writeGauge` also write the histogram percentiles?

Reply:
Yes, I've included tests on percentiles to verify it.

Reply:
Histogram objects usually also include count. Is there an easy way to print the count per histogram as well?

Instead of tracking throughput independently they can be derived from the histgorams.